routing_algorithms/many_to_many.hpp

```diff
@@ -113,7 +114,7 @@ class ManyToManyRouting final
 
         // for each source do forward search
         unsigned source_id = 0;
-        for (const std::vector<PhantomNode> &phantom_node_vector : phantom_nodes_array)
+        for (const std::vector<PhantomNode> &phantom_node_vector : (phantom_sources_array.size() ? phantom_sources_array : phantom_targets_array))
```

Comment:
This should not have a switch. Always use `phantom_sources_array`. Caller needs to take care to populate both arrays accordingly.